03_python_fastapi_project/main.py

```
@@ -144,6 +162,61 @@ async def get_product(product_id: int, db: AsyncSession = Depends(get_db)):
+    else:
+        db_cartitem = CartItem(product_id=product_id, quantity=1)
+        db.add(db_cartitem)
+    # Decrement stock
+    db_product.stock -= 1
+    await db.commit()
+    await db.refresh(db_cartitem)
+    return cartitem_to_response(db_cartitem)
+
+
+@app.post("/cart/remove/{cart_item_id}", response_model=CartItemResponse)
+async def remove_from_cart(cart_item_id: int, db: AsyncSession = Depends(get_db)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)